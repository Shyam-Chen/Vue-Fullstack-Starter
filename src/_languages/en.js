export default {
  /** @name practical-examples */
  practicalExamples: 'Practical Examples',

  overview: 'Overview',
  examples: 'Examples',

  helloWorld: 'Hello World',

  guide: 'Guide',
  templateRendering: 'Template Rendering',
  components: 'Components',
  props: 'Props',
  slots: 'Slots',
  refs: 'Refs',
  customEvents: 'Custom Events',
  extends: 'Extends',
  dynamic: 'Dynamic',
  recursion: 'Recursion',
  lifecycleHooks: 'Lifecycle Hooks',
  transitions: 'Transitions',
  mixins: 'Mixins',
  directives: 'Directives',
  dependencyInjection: 'Dependency Injection',
  filters: 'Filters',
  routing: 'Routing',
  dynamicMatching: 'Dynamic Matching',
  nestedRoutes: 'Nested Routes',
  guards: 'Guards',
  stateManagement: 'State Management',

  crudOperations: 'CRUD Operations',
  basic: 'Basic',
  rest: 'REST',
  graphql: 'GraphQL',

  controls: 'Controls',
  textFields: 'Text Fields',
  selects: 'Selects',
  selectionControls: 'Selection Controls',
  pickers: 'Pickers',
  autocomplete: 'Autocomplete',
  stepper: 'Stepper',
  tabs: 'Tabs',
  treeView: 'Tree View',
  calendar: 'Calendar',
  markdownEditor: 'Markdown Editor',
  richTextEditor: 'Rich Text Editor',

  dataTable: 'Data Table',
  displayRows: 'Display Rows',
  inlineEdit: 'Inline Edit',
  paginator: 'Paginator',
  iterator: 'Iterator',
  exportReport: 'Export Report',

  internationalization: 'Internationalization',

  dataVisualization: 'Data Visualization',

  authorization: 'Authorization',

  payment: 'Payment',

  messages: 'Messages',
  chatRoom: 'Chat Room',
  email: 'Email',
  cloudMessaging: 'Cloud Messaging',

  machineLearning: 'Machine Learning',

  /** @name additional-resources */
  additionalResources: 'Additional Resources',

  universalApps: 'Universal Apps',
  mobileApps: 'Mobile Apps',
  desktopApps: 'Desktop Apps',
};
