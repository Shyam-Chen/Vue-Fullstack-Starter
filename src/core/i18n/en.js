import app from '~/_languages/en';
import home from '~/home/_languages/en';
import guideStateManagement from '~/shell/guide/state-management/_languages/en';
import internationalization from '~/shell/internationalization/_languages/en';

export default {
  ...app,
  home,
  ...guideStateManagement,
  ...internationalization,
};
