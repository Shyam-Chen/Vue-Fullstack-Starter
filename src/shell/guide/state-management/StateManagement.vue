<template>
  <v-container id="state-management" fluid>
    <v-layout column>
      <v-layout row>
        <div class="headline pa-2">
          Clicked: <strong>{{ c$.value }}</strong> times, value is <strong>{{ evenOrOdd }}</strong
          >.
        </div>
      </v-layout>

      <v-layout row>
        <v-subheader class="pt-3">Step:</v-subheader>
        <div>
          <v-select v-model="c$.step" :items="c$.stepList" class="step" single-line></v-select>
        </div>
      </v-layout>

      <v-layout row>
        <v-btn color="primary" @click="increment">{{ $t('increment') }}</v-btn>
        <v-btn color="primary" @click="decrement">{{ $t('decrement') }}</v-btn>
      </v-layout>

      <v-layout row>
        <v-btn color="primary" @click="incrementAsync"
          >{{ $t('increment') }} ({{ $t('async') }})</v-btn
        >
        <v-btn color="primary" @click="decrementAsync"
          >{{ $t('decrement') }} ({{ $t('async') }})</v-btn
        >
        <v-btn color="primary" @click="incrementIfOdd"
          >{{ $t('increment') }} ({{ $t('ifOdd') }})</v-btn
        >
        <v-btn color="primary" @click="decrementIfEven"
          >{{ $t('decrement') }} ({{ $t('ifEven') }})</v-btn
        >
      </v-layout>

      <v-layout row>
        <v-btn color="primary" disabled>Start Count</v-btn>
        <v-btn color="primary" disabled>Cancel Count</v-btn>
      </v-layout>
    </v-layout>
  </v-container>
</template>

<script>
// @flow

import { mapActions, mapGetters } from 'vuex';

import { ICounter } from './types';
import { INITIAL as state } from './constants';
import actions from './actions';
import mutations from './mutations';
import getters from './getters';

export default {
  metaInfo: {
    title: 'State Management | Oh My Vue',
  },
  computed: {
    c$(): ICounter {
      return this.$store.state.counter;
    },
    ...mapGetters('counter', Object.keys(getters)),
  },
  created() {
    this.$store.registerModule('counter', { namespaced: true, state, actions, mutations, getters });
  },
  methods: {
    ...mapActions('counter', Object.keys(actions)),
  },
};
</script>

<style scoped>
.step {
  width: 5rem;
}
</style>
