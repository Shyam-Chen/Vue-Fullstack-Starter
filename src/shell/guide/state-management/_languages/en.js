export default {
  increment: 'Increment',
  decrement: 'Decrement',
  async: 'Async',
  ifOdd: 'If Odd',
  ifEven: 'If Even',
};
