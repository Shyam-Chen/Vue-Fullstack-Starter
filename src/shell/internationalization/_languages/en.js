export default {
  bubble: 'Bubble',
  hello: 'Hello, {msg}!',
  thinkIn: 'Think in {2}',
  http: 'http | https',
  framework: 'No framework | One framework | {num} frameworks',
};
