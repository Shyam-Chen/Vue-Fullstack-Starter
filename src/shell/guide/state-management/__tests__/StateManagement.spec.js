import Vuex from 'vuex';
import { createLocalVue, shallowMount } from '@vue/test-utils';
import I18n from 'vue-i18n';

import StateManagement from '../StateManagement';
import en from '../_languages/en';

const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(I18n);

describe('StateManagement', () => {
  let [wrapper, store, i18n] = [];
  let [state, actions, mutations, getters] = [];

  beforeEach(() => {
    state = { value: 0 };

    actions = {
      increment: jest.fn(),
      decrement: jest.fn(),
      incrementAsync: jest.fn(),
      decrementAsync: jest.fn(),
      incrementIfOdd: jest.fn(),
      decrementIfEven: jest.fn(),
    };

    mutations = {
      increment: jest.fn(),
      decrement: jest.fn(),
    };

    getters = {
      evenOrOdd: jest.fn(() => 'even'),
    };

    store = new Vuex.Store({
      modules: {
        counter: {
          state,
          actions,
          mutations,
          getters,
        },
      },
    });

    i18n = new I18n({
      locale: 'en',
      messages: { en },
    });

    wrapper = shallowMount(StateManagement, { localVue, store, i18n });
  });

  it('should render an initial component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
