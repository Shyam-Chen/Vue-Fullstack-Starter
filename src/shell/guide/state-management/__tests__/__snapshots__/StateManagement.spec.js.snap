// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StateManagement should render an initial component 1`] = `
<v-container-stub id="state-management" tag="div" fluid="true" class="shell-container">
  <v-layout-stub tag="div" column="">
    <v-layout-stub tag="div" row="">
      <div class="headline pa-2">
        Clicked: <strong>0</strong> times, value is <strong>even</strong>.
      </div>
    </v-layout-stub>
    <v-layout-stub tag="div" row="">
      <v-subheader-stub class="pt-3">Step:</v-subheader-stub>
      <div>
        <v-select-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" value="1" appendicon="$dropdown" backgroundcolor="" loaderheight="2" clearicon="$clear" singleline="true" type="text" valuecomparator="function deepEqual(a, b) {
  if (a === b) return true;

  if (a instanceof Date &amp;&amp; b instanceof Date) {
    // If the values are Date, they were convert to timestamp with getTime and compare it
    if (a.getTime() !== b.getTime()) return false;
  }

  if (a !== Object(a) || b !== Object(b)) {
    // If the values aren't objects, they were already checked for equality
    return false;
  }

  var props = Object.keys(a);

  if (props.length !== Object.keys(b).length) {
    // Different number of props, don't bother to check
    return false;
  }

  return props.every(function (p) {
    return deepEqual(a[p], b[p]);
  });
}" nodatatext="$vuetify.noDataText" items="1,2,3,4,5,6,7,8,9,10" itemcolor="primary" itemdisabled="disabled" itemtext="text" itemvalue="value" menuprops="[object Object]" class="step"></v-select-stub>
      </div>
    </v-layout-stub>
    <v-layout-stub tag="div" row="">
      <v-btn-stub color="primary" tag="button" activeclass="" type="button">Increment</v-btn-stub>
      <v-btn-stub color="primary" tag="button" activeclass="" type="button">Decrement</v-btn-stub>
    </v-layout-stub>
    <v-layout-stub tag="div" row="">
      <v-btn-stub color="primary" tag="button" activeclass="" type="button">Increment (Async)</v-btn-stub>
      <v-btn-stub color="primary" tag="button" activeclass="" type="button">Decrement (Async)</v-btn-stub>
      <v-btn-stub color="primary" tag="button" activeclass="" type="button">Increment (If Odd)</v-btn-stub>
      <v-btn-stub color="primary" tag="button" activeclass="" type="button">Decrement (If Even)</v-btn-stub>
    </v-layout-stub>
    <v-layout-stub tag="div" row="">
      <v-btn-stub color="primary" tag="button" activeclass="" disabled="true" type="button">Start Count</v-btn-stub>
      <v-btn-stub color="primary" tag="button" activeclass="" disabled="true" type="button">Cancel Count</v-btn-stub>
    </v-layout-stub>
  </v-layout-stub>
</v-container-stub>
`;
